### PR TITLE
Fixed RTL support

### PR DESCRIPTION
**Pull request by reseul**
We rely on Xaml FlowDirection rather than on the Yoga based flipping.
We also support changing the direction on the fly.
Various peculiarities/details can be found in the comments throughout the PR.

### CODE REVIEW

File: ReactWindows/ReactNative.Net46/Views/Text/ReactTextShadowNode.cs [hunk 1]
```diff
@@ -254,8 +254,8 @@ private void UpdateTextBlockCore(TextBlock textBlock, bool measureOnly)
                 textBlock.Padding = new Thickness(
                     GetPadding(YogaEdge.Left),
                     GetPadding(YogaEdge.Top),
-                    0,
-                    0);
+                    GetPadding(YogaEdge.Right),
```
Comment (rozele) [inline ReactWindows/ReactNative.Net46/Views/Text/ReactTextShadowNode.cs]: Interesting - is this an unrelated bug?

File: ReactWindows/ReactNative.Shared/UIManager/UIImplementation.cs [hunk 2]
```diff
@@ -149,6 +149,19 @@ public Task RemoveRootViewAsync(int rootViewTag)
...
         }
 
+        /// <summary>
+        /// Refreshes RTL/LTR direction on all root views.
+        /// </summary>
+        public void UpdateLayoutDirection()
+        {
+            // RTL/LTR is implemented through Xaml FlowDirection, so we keep Yoga oblivious of this.
+            _operationsQueue.UpdateRootViewNodesDirection();
+            if (_operationsQueue.IsEmpty())
```
Comment (rozele) [inline ReactWindows/ReactNative.Shared/UIManager/UIImplementation.cs]: What is the purpose of calling `DispatchViewUpdates` if `_operationsQueue.IsEmpty()`?

File: ReactWindows/ReactNative.Net46/Views/Text/ReactTextShadowNode.cs (hunk 1 above)
Comment (reseul) [reply ReactWindows/ReactNative.Net46/Views/Text/ReactTextShadowNode.cs]: This was the reason the text padding was not working correctly in RTL. I used fuzzy logic (i.e. without clear understanding on how it's supposed to work) and made the thing consistent by not ignoring some values. The padding error was suddenly fixed...

---
In reply to: [223475202](https://github.com/Microsoft/react-native-windows/pull/1994#discussion_r223475202) [](ancestors = 223475202)

File: ReactWindows/ReactNative.Shared/UIManager/UIImplementation.cs (hunk 2 above)
Comment (reseul) [reply ReactWindows/ReactNative.Shared/UIManager/UIImplementation.cs]: I also followed the model from the previous method, theoretically taking into account the case this update comes outside any UIManager batch. It's just supposed to flush this change to the operation queues in that case.

---
In reply to: [223475746](https://github.com/Microsoft/react-native-windows/pull/1994#discussion_r223475746) [](ancestors = 223475746)

**Pull request merged**